aiohttp/web.py

```diff
@@ -1436,55 +1437,48 @@ def _register_endpoint(self, route):
 
     def add_route(self, method, path, handler,
                   *, name=None, expect_handler=None):
-        assert path.startswith('/')
+
         assert callable(handler), handler
         if not asyncio.iscoroutinefunction(handler):
             handler = asyncio.coroutine(handler)
         method = method.upper()
         assert method in self.METHODS, method
-        parts = []
-        factory = PlainRoute
-        format_parts = []
-        for part in path.split('/'):
-            if not part:
-                continue
+
+        if not ('{' in path or '}' in path or self.ROUTE_RE.search(path)):
+            route = PlainRoute(
+                method, handler, name, path, expect_handler=expect_handler)
+            self._register_endpoint(route)
+            return route
+
+        pattern = ''
+        formatter = ''
+        for part in self.ROUTE_RE.split(path):
             match = self.DYN.match(part)
             if match:
-                parts.append('(?P<' + match.group('var') + '>' +
-                             self.GOOD + ')')
-                factory = DynamicRoute
-                format_parts.append('{'+match.group('var')+'}')
+                pattern += '(?P<' + match.group('var') + '>' + self.GOOD + ')'
+                formatter += '{%s}' % match.group('var')
```

Comment:
Please use `.format()` for string formatting: other parts of aiohttp tends to `%` style only (but we have couple exceptions now).